src/pages/examples/spatial-queries.hbs

```diff
@@ -97,36 +95,40 @@ layout: example.hbs
 
   // get references to our <select> elements
   var relationship = document.getElementById('relationSelect');
-  var geometry = document.getElementById('geometrySelect');
+  var geometry = document.getElementById('geometrySelect');  
+  var runQueryButton = document.getElementById('executeQuery');
 
   var previousIds = [];
 
   // reset all features back to their regularly defined styles
   function reset(){
     for (var i = previousIds.length - 1; i >= 0; i--) {
       neighborhoods.resetStyle(previousIds[i]);
-    };
+    };   
   }
 
   // query the API and highlight features
   function query(){
-    reset();
+    reset();    
 
     // lookup our input geometry
     var inputGeometry = geometries[geometry.value];
 
     // query the service executing the selected relation with the selected input geometry
     neighborhoods.query()[relationship.value](inputGeometry).ids(function(error, ids){
-      previousIds = ids;
-      for (var i = ids.length - 1; i >= 0; i--) {
-        neighborhoods.setFeatureStyle(ids[i], { color: 'red', weight: 2 });
-      };
+      if (error) {
+        console.log('Error with query: ' + error);
+      } else {
```

Comment:
to account for when `ids` can be `null` instead of an array
```suggestion
      } else if (ids) {
```

Reply:
I've made updates.  Please review again.  I sometimes have trouble with removing trailing spaces, so hopefully that is resolved before.

And yeah, the errors with the null ids were present before.  But I think your suggestion resolves that issue as well.